### ISSUE TITLE
Stop-Computer, Restart-Computer on Unix: inability to detect failure.

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#19802]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

On Unix, these commands defer to the native `/sbin/shutdown` utility.

However, the way in which this is currently implemented makes it impossible to detect failure _programmatically_:

E.g., on macOS, _without_ using `sudo`, run the following:

```
# Run *without* sudo, so as to provoke failure of /sbin/shutdown
Stop-Computer -ErrorVariable e 2>$null || Write-Error "Shutdown initiation failed."
```

Note that running `/sbin/shutdown` directly in the scenario above:

* prints `shutdown: NOT super-user` to stderr
* reports `1` as its exit code

### Expected behavior

No visible output (due to `2>$null`), the error message stored in `$e`, and `$?` indicating `$false` so that the `||` clause is triggered.

### Actual behavior

`shutdown`'s stderr prints directly to the terminal, bypassing all PowerShell streams.
`$?` reflects `$true` afterwards.

In other words: there is no way to detect failure.

### Environment data

```powershell
PowerShell 7.4.0-preview.3 on macOS and Linux
```
